tests/lib/testing_classes.py

```diff
@@ -331,19 +335,30 @@ def publish_finished(self, dbcon, launched_app, download_test_data,
             print("Creating only setup for test, not launching app")
             yield False
             return
-        import time
+
         time_start = time.time()
         timeout = timeout or self.TIMEOUT
         timeout = float(timeout)
+
+        output = ""
+        errors = ""
         while launched_app.poll() is None:
-            time.sleep(0.5)
-            if time.time() - time_start > timeout:
-                launched_app.terminate()
-                raise ValueError("Timeout reached")
+            try:
+                out, err = launched_app.communicate(timeout=0.5)
+            except subprocess.TimeoutExpired:
+                out, err = launched_app.communicate()
+                if time.time() - time_start > timeout:
```

Comment:
I don't understand this condition. `out, err = launched_app.communicate()` won't stop until the process finishes...

In other words the timeout is broken now...

Reply:
Something like this could work...
```python
out = b""
err = b""
while launched_app.poll() is None:
    if launched_app.stdout is not None:
        out += launched_app.stdout.read()

    if launched_app.stderr is not None:
        err += launched_app.stderr.read()

    if time.time() - time_start > timeout:
        launched_app.terminate()
        raise ValueError("Timeout reached")
```

Reply:
This was mostly driven by @BigRoy, so will leave that up to you two.

Reply:
Thanks @iLLiCiTiT have used this to resolve the Nuke issue mentioned.